src/Features/CSharp/Portable/InlineParameterNameHints/InlineParamNameHintsService.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineParameterNameHintsService;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.InlineParameterNameHints
+{
+    [ExportLanguageService(typeof(IInlineParamNameHintsService), LanguageNames.CSharp), Shared]
+    internal class InlineParamNameHintsService : IInlineParamNameHintsService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InlineParamNameHintsService()
+        {
+        }
+
+        public async Task<IEnumerable<(string, TextSpan)>> GetInlineParameterNameHintsAsync(
+            Document document,
+            TextSpan textSpan,
+            CancellationToken cancellationToken)
+        {
+            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var node = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+            var spans = new List<(string name, TextSpan span)>();
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var invocations = node.DescendantNodes().OfType<InvocationExpressionSyntax>();
+            foreach (var invo in invocations)
+            {
+                foreach (var argument in invo.ArgumentList.Arguments)
+                {
+                    if (argument.NameColon == null && IsLiteralOrNoNamedExpression(argument))
+                    {
+                        var param = argument.DetermineParameter(semanticModel, cancellationToken: cancellationToken);
+                        if (param != null)
+                        {
+                            spans.Add((param.Name, argument.Span));
+                        }
+                    }
+                }
+            }
+
+            return spans;
+        }
+
+        private static bool IsLiteralOrNoNamedExpression(ArgumentSyntax arg)
+        {
+            if (arg.Expression is LiteralExpressionSyntax)
+            {
+                return true;
+            }
+            if (arg.Expression is ObjectCreationExpressionSyntax)
+            {
+                return true;
+            }
+            if (arg.Expression is CastExpressionSyntax cast)
+            {
+                // Determine if the descendant node from the cast is of literal type
+                // If so, then we should add the adornment
+                var literal = arg.DescendantNodes().OfType<LiteralExpressionSyntax>();
```

Comment:
This will be looking at all descendants, so I imagine something like OuterCall(InnerCall(true)) would show parameters for both which is probably not what you intended.

Reply:
Changed the logic